Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/SqlDialectTest.cs

```diff
@@ -49,5 +57,219 @@ public void GetParameterName ()
       Assert.That(_dialect.GetParameterName("parameter"), Is.EqualTo("@parameter"));
       Assert.That(_dialect.GetParameterName("@parameter"), Is.EqualTo("@parameter"));
     }
+
+    [Test]
+    [CLSCompliant(false)]
+    [TestCase(null, TestName = "CreateDataParameter_WithoutSize_DoesNotSetSizeOnParameter.")]
+    [TestCase(-1, TestName = "CreateDataParameter_WithNegativeSize_SetsSizeOnParameter.")]
+    [TestCase(0, TestName = "CreateDataParameter_WithSizeZero_SetsSizeOnParameter.")]
+    [TestCase(1, TestName = "CreateDataParameter_WithPositiveSize_SetsSizeOnParameter.")]
+    public void CreateDataParameter (int? storageTypeSize)
+    {
+      var commandMock = new Mock<IDbCommand>(MockBehavior.Strict);
+      var dataParameterMock = new Mock<IDbDataParameter>(MockBehavior.Strict);
+
+      var storageTypeInformation = new StorageTypeInformation(
+          typeof(bool),
+          "test",
+          DbType.Boolean,
+          false,
+          storageTypeSize,
+          typeof(int),
+          _typeConverterStub.Object);
+
+      _typeConverterStub.Setup(_ => _.ConvertTo(null, null, "value", storageTypeInformation.StorageType)).Returns("");
+
+      commandMock.Setup(_ => _.CreateParameter()).Returns(dataParameterMock.Object).Verifiable();
+
+      dataParameterMock.SetupSet(_ => _.ParameterName = "@dummy").Verifiable();
+      dataParameterMock.SetupSet(_ => _.DbType = storageTypeInformation.StorageDbType).Verifiable();
+      dataParameterMock.SetupSet(_ => _.Value = "").Verifiable();
+      if (storageTypeSize.HasValue)
+        dataParameterMock.SetupSet(_ => _.Size = storageTypeSize.Value).Verifiable();
+
+      var result = _dialect.CreateDataParameter(commandMock.Object, storageTypeInformation, "@dummy", "value");
+
+      commandMock.Verify();
+      dataParameterMock.Verify();
+
+      Assert.That(result, Is.SameAs(dataParameterMock.Object));
+      if (!storageTypeSize.HasValue)
+        dataParameterMock.VerifySet(_ => _.Size = It.IsAny<int>(), Times.Never());
+    }
+
+    [Test]
+    public void CreateDataParameter_WithStringValueExceedingFixedSize_DoesNotInitializeSize ()
```

Comment:
is there a reason why you do this via three tests instead of just using a TestCase or TestCaseSource? Since you're using testcase for the previous test it feels weird to not do it here, too for string/char[]/byte[]

Reply:
The reason is that we are dealing with different types here, which affects the original value, the converted value, and the `StorageTypeInformation`. It is more readable as three separate test cases.